app/private_publisher.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+#   coding: UTF-8
+
+from app.queue_publisher import QueuePublisher
+
+from cryptography.fernet import Fernet
+
+
+class PrivatePublisher(QueuePublisher):
+
+    @staticmethod
+    def encrypt(message, secret):
+        """
+        Message may be a string or bytes.
+        Secret key must be 32 url-safe base64-encoded bytes.
+
+        """
+        try:
+            f = Fernet(secret)
+        except ValueError:
+            return None
+        try:
+            token = f.encrypt(message)
+        except TypeError:
+            token = f.encrypt(message.encode("utf-8"))
+        return token
+
+    @staticmethod
+    def decrypt(token, secret):
+        """
+        Secret key must be 32 url-safe base64-encoded bytes or string.
+
+        Returned value is a string.
+        """
+        try:
```

Comment:
They could be combined, but maybe at the expense of readability. The ValueError is explicitly from trying to make a Fernet object with an invalid secret. That's non-recoverable.
The TypeError is from trying to encrypt a non-bytes object. That's recoverable; we can helpfully assume we've been passed a string (most likely the case) and try again. So they are both handling exceptions, but the one is an error condition and the other not.